src/Workspaces/Core/Portable/Workspace/Host/TemporaryStorage/TemporaryStorageHandle.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading;
+
+namespace Microsoft.CodeAnalysis.Host;
+
+/// <summary>
+/// Represents a handle to data stored to temporary storage (generally a memory mapped file).  As long as this handle is
+/// not disposed, the data should remain in storage and can be readable from any process using the information provided
+/// in <see cref="Identifier"/>.  Use <see cref="ITemporaryStorageServiceInternal.WriteToTemporaryStorage"/> to write
+/// the data to temporary storage and get a handle to it.  Use <see
+/// cref="ITemporaryStorageServiceInternal.ReadFromTemporaryStorageService"/> to read the data back in any process.
+/// </summary>
+internal sealed class TemporaryStorageHandle(IDisposable underlyingData, TemporaryStorageIdentifier identifier) : IDisposable
```

Comment:
note: while this si disposable.  only one component in all of our system actually disposes these.  And that's just for the project-system compiler-command-line-args case.  All the places we use MMFs for references, we never dispose.  I'm considering breaking things out into different APIs.  one used by references (with no handles, and no dispose) and one used by the proejct system.